### PR TITLE
[#713] Process L and R separatly, ignore unsupported pages

### PR DESCRIPTION
**Pull request by olaurendeau**
- [x] Process L and R tag separatly
- [x] Ignore unsupported pages

capture d ecran 2017-06-04 a 10 25 07

### CODE REVIEW

File: c2corg_ui/format/ltag.py [hunk 1]
```diff
@@ -57,23 +65,44 @@ class LTagProcessor(BlockProcessor):
 
     RE_LTAG_BLOCK = re.compile(r'^\s*(?:[L|R]#(?:[^|\n]*(?:\|[^|\n]*)+)\n?)+$')
     RE_LTAG = re.compile(r'([L|R])#(_|~|[0-9]*)')
+    RE_RTAG_ROW_DISTINCTION = re.compile(r'^\s*R#.*$')
     RE_PIPE_SELECTION = re.compile(r'(\|)(?![^|]*\]\])')
     ltag_template = '<span class="pitch"><span translate>{0}</span>{1}</span>'
     pipe_replacement = '__--|--__'
-    row_count = 1
+    pitch_count = 1
+    relay_count = 1
```
Comment (asaunier) [inline c2corg_ui/format/ltag.py]: Isn't "relay" a bit confusing? I think R rather stands for "rappel".

File: c2corg_ui/format/ltag.py [hunk 2]
```diff
@@ -16,9 +16,14 @@
 class LTagPreprocessor(Preprocessor):
     """ Preprocess lines to clean LTag blocks """
 
+    RE_LTAG_UNSUPPORTED = re.compile(r'[L|R]#[^\s|]+')
```
Comment (asaunier) [inline c2corg_ui/format/ltag.py]: Could you add a comment to precise what L# syntaxes are actually supported (and what are not)?

File: c2corg_ui/format/ltag.py [hunk 3]
```diff
@@ -57,23 +65,44 @@ class LTagProcessor(BlockProcessor):
...
+        if col_number == 1 and pitch_type == "R":
+            return self.relay_count
+
+        return self.pitch_count
 
     def is_ltag(self, text):
         return self.RE_LTAG_BLOCK.search(text) is not None
 
     def test(self, parent, block):
         """  """
```
Comment (asaunier) [inline c2corg_ui/format/ltag.py]: Is this empty comment line really required?

File: c2corg_ui/format/ltag.py [hunk 4]
```diff
@@ -57,23 +65,44 @@ class LTagProcessor(BlockProcessor):
...
+
+    def increment_row_count(self, row):
+        if self.RE_RTAG_ROW_DISTINCTION.search(row) is not None:
+            self.relay_count += 1
+            return
+
+        self.pitch_count += 1
+
+    def get_row_count(self, col_number, pitch_type):
+
```
Comment (asaunier) [inline c2corg_ui/format/ltag.py]: Perhaps avoid unconsistent blank lines as this one

File: c2corg_ui/format/ltag.py [hunk 5]
```diff
@@ -16,18 +16,27 @@
 class LTagPreprocessor(Preprocessor):
     """ Preprocess lines to clean LTag blocks """
 
+    """ Supported tags are L#, R# """
+    RE_LTAG_UNSUPPORTED = re.compile(r'[L|R]#[^\s|]+')
```
Comment (asaunier) [inline c2corg_ui/format/ltag.py]: Shouldn't this be ``RE_LTAG_SUPPORTED``?

Comment (olaurendeau) [inline c2corg_ui/format/ltag.py]: Nop this regex is checking in the whole block if an **un**supported tag is present :)

Comment (asaunier) [inline c2corg_ui/format/ltag.py]: I am not sure to understand your regexp then: doesn't it catch strings such as ``L# |`` or ``R# |``?

Comment (olaurendeau) [inline c2corg_ui/format/ltag.py]: See https://regex101.com/r/sxCfUO/2 for examples
It catch every other tags :)

**Comment by olaurendeau** [comment]
Can I merge it or should it be various review & approvals ?

**Comment by stef74** [comment]
If @asaunier approuve you can merge

**Comment by desnoes** [comment]
@olaurendeau : it seems that the last PR has broken the parsing of L# tags. At least almost 100% of climbing routes with L# are not parsed correctly. Could you have a look ? We have blocked the release of the new version in order to fix this bug.

**Pull request merged**